### ISSUE TITLE
Power management signals

### ISSUE THREAD

**Issue by shermp** [shermp/NickelDBus#16]
There were discussions on https://github.com/pgaskin/NickelMenu/issues/124 about using the signals from `N3PowerWorkflowManager` to notify scripts/programs when resuming from sleep etc. It's probably best to continue the discussion over here.

For reference, the available signals are:
```
aboutToSleep()
resumingFromSleep()
resumedFromSleep()
aboutToPowerOff()
aboutToReboot() 
```

Pinging @marnovdm @NiLuJe

**Comment by marnovdm** [comment]
Yes this seems like a better place, thanks for opening this! I (unintentionally) hijacked the original issue post a bit anyway :smile: 

Not sure if I can contribute anything here, as I'm not really familiar with C++, and I'll be on holiday for the next couple of weeks so it'll be a while before I get back to this issue.

**Comment by marnovdm** [comment]
It's been a while but like I said I'm not familiar with C++. I've been testing with Claude Code recently and had some extra credits to burn so decided to let it take a shot at this, and it actually really impressed me by basically one-shotting this feature in NickelMenu repo. It recommended an eventFilter approach in nickelmenu.cc to easily detect wake events:

```
 // Wake event detection: Install Qt event filter to monitor application
 // activation events. When Nickel resumes from sleep, Qt fires
 // ApplicationActivate events. We use this to trigger wake hooks.
```

If there is interest I can try and open a pull request for these changes, but I wanted to check here first because I know not all projects appreciate AI PRs like this. It works for me but I'm very much aware that I can't review the quality of these changes.

**Comment by busybox11** [comment]
Doesn't sound too unreasonable!
As a general rule of thumb, even though AI as a whole is not my cup of tea, I believe that as long as you use it responsibly and you know what you're doing with it - that is, not submitting random, untested, unreviewed PRs - I don't see any downside (other than LLMs being the thing that they are).

It can be an incredible tool, and I could see it come in really handy for speeding things up in embedded systems modding, like kobo, which don't have a huge community but could grow quite substantially if used properly.
